src/main/java/org/jenkinsci/plugins/workflow/graph/FlowNode.java

```diff
@@ -248,6 +248,25 @@ protected void setActions(List<Action> actions) {
         this.actions = new CopyOnWriteArrayList<Action>(actions);
     }
 
+
+    /**
+     * Return a persistent action o(transient actions are not included)
+     * @param type Action type to look for
+     * @param <T> Action type to look for
+     * @return First attached action of given type, or null if none found
+     */
+    public <T extends Action> T getDirectAction(Class<T> type) {
+        // Need to check and see if this will cause issues -- better to change in place if we can
+        // Otherwise we need to do changes in many places to use a new, fast API
+        List<Action> acts = (actions == null) ? getActions() : actions;
```

Comment:
@jglick After digging in again on this, I've done the Iterator optimization (commit pending once I finish reverting the older code and benchmarking) as a mini-optimization.  

Since the field is only modified by `getActions( )` and `setActions(List<Action> newActions)`  (and field assignment is atomic, and in the null case we invoke getActions to for the same result), the result *should* be thread safe AFAICT.